docs/changelog/109173.yaml

```diff
@@ -0,0 +1,5 @@
+pr: 109173
+summary: Wrap "Pattern too complex" exception into an `IllegalArgumentException`
```

Comment:
Did this go live in an Elasticsearch release? If not I don't think it needs a changelog.